docs/api-guide/routers.md

```diff
@@ -72,14 +72,23 @@ Alternatively you can use Django's `include` function, like so…
         url(r'^', include(router.urls)),
     ]
 
-Router URL patterns can also be namespaces.
+Router URL patterns can also be included using _application_ and _instance_ namespaces.
+
+To use an _application namespace_ pass a 2-tuple containing the `router.urls` and the app name.
+To use an _instance namespace_ pass the optional `namespace` parameter.
```

Comment:
Can we reference anywhere in the Django docs here? I wouldn't know the difference between the two off the top of my head.

I guess ideally we'd just say "Do X", and have "If you need to do Y then ..." slightly later in the docs, rather than presenting the developer with a choice that they need to make here.

Reply:
The `[URL namespaces docs]` reference below is the canonical source. It's basically the only place it's ever discussed.

The other place we could link to is the `include` reference https://docs.djangoproject.com/en/2.0/ref/urls/#include

The way I thought about phrasing this is: 

1. Try to `include` via an app-level `urls` module, setting the `app_name` variable there. Basically never use the `namespace` parameter, you basically don't want it. 
2. If you MUST `include` the router urls directly then provide the `(router.urls, app_name)` — that's what you want.  (If you really need `namespace` then go-ahead, but you still have to pass the 2-tuple first.) 

But by the time I'd drafted than I basically need to rewrite the Django docs (linked) so people should really use those.

Reply:
The >80% case is to just pass the 2-tuple and ignore the `namespace` parameter. Let me re-phrase around that...